Shapes/Triangle.cs

```diff
@@ -0,0 +1,62 @@
+﻿namespace Shapes
+{
+    public class Triangle : Shape
+    {
+        private int Height;
+
+        public Triangle(int height, bool isFilled, char sign) : base(isFilled, sign)
+        {
+            this.Height = height;
+        }
+
+        protected override void PrintFilled()
+        {
+            int baseLength = this.Height * 2 - 1;
+
+            for (int heightIndex = 0; heightIndex < this.Height; heightIndex++)
+            {
+                for (int baseIndex = 0; baseIndex < baseLength; baseIndex++)
+                {
+                    if (baseIndex >= (baseLength - (2 * heightIndex + 1)) / 2 && baseIndex <= (baseLength + (2 * heightIndex + 1)) / 2)
+                    {
+                        Console.Write($"{Sign} ");
+                    }
+                    else
+                    {
+                        Console.Write("  ");
+                    }
+                }
+                Console.WriteLine();
+            }
+        }
+
+        protected override void PrintNonFilled()
+        {
+            int baseLength = this.Height * 2 - 1;
+
+            for (int heightIndex = 0; heightIndex < this.Height; heightIndex++)
+            {
+                for (int baseIndex = 0; baseIndex < baseLength; baseIndex++)
+                {
+                    if (heightIndex == this.Height - 1 || baseIndex == (baseLength - (2 * heightIndex + 1)) / 2 || baseIndex == (baseLength + (2 * heightIndex + 1)) / 2)
```

Comment:
One of many places with the same comment(a function to hold a long and tiring if statement).